src/Microsoft.DotNet.Darc/DarcLib/GitHubClient.cs

```diff
@@ -733,7 +733,19 @@ private async Task<GitFile> GetGitItemImpl(string path, TreeItem treeItem, strin
         using (HttpClient client = CreateHttpClient())
         {
             var requestManager = new HttpRequestManager(client, method, requestUri, logger, body, versionOverride, logFailure);
-            return await requestManager.ExecuteAsync(retryCount);
+            try
+            {
+                return await requestManager.ExecuteAsync(retryCount);
+            }
+            catch (HttpRequestException ex) when (ex.StatusCode == HttpStatusCode.Forbidden)
+            {
+                if (logFailure)
+                {
+                    _logger.LogError("Check SAML single sign-on (SSO) for your token: " +
```

Comment:
So this can be even the token not being set at all. So this error is a symptom for more possible cases.

It would be better if this said something like "It seems like your authentication is misconfigured.. blabla .. please also check that the GitHub token you have provided is SSO enabled for the organization of the repository".

Instead of the aka.ms link, we should:
- Mention that `darc authenticate` is the command to fix this
- Link to https://github.com/dotnet/arcade/blob/main/Documentation/Darc.md#step-3-set-additional-pats-for-azure-devops-and-github-operations
- Put the SSO instructions with the aka.ms link, possibly with the screenshot from issue, in the `Darc.md` documentation.

Reply:
What do you think:
"""
It appears that there is a misconfiguration in your authentication. Please check the documentation at https://github.com/dotnet/arcade/blob/main/Documentation/Darc.md#step-3-set-additional-pats-for-azure-devops-and-github-operations. Additionally, ensure that the GitHub token you have supplied is Single Sign-On (SSO) enabled for the organization associated with the repository.
"""

Reply:
Great. Maybe a bit more concise version?

> Your GitHub token seems to be invalid. Please see https://github.com/dotnet/arcade/blob/main/Documentation/Darc.md?rgh-link-date=2024-01-11T14%3A26%3A02Z#step-3-set-additional-pats-for-azure-devops-and-github-operations
> Make sure the GitHub token is Single Sign-On (SSO) enabled for the organization associated with the repository.